File: surface/widgets/gtk/components/widgets/feedback/AlertDialog.ts
// ============================================================
// Clef Surface GTK Widget — AlertDialog
//
// Confirmation dialog with title, description, and action
// buttons. Uses Adw.MessageDialog for native GNOME alert
// dialog presentation.
//
// Adapts the alert-dialog.widget spec: anatomy (root, backdrop,
// content, title, description, actions, cancel, action), states
// (open, closed), and connect attributes to GTK4/Adwaita
// rendering.
// ============================================================

import Gtk from 'gi://Gtk?version=4.0';
import Adw from 'gi://Adw?version=1';

// --------------- Props ---------------

export interface AlertDialogProps {
  open?: boolean;
  title?: string;
  description?: string;
  confirmLabel?: string;
  cancelLabel?: string;
  destructive?: boolean;
  parent?: Gtk.Window | null;
  onConfirm?: () => void;
  onCancel?: () => void;
}

// --------------- Component ---------------

/**
 * Creates an Adwaita MessageDialog for confirmation prompts with
 * configurable title, description, and action buttons.
 */
export function createAlertDialog(props: AlertDialogProps = {}): Gtk.Widget {
  const {
    title = 'Confirm',
    description = '',
    confirmLabel = 'OK',
    cancelLabel = 'Cancel',
    destructive = false,
    parent = null,
    onConfirm,
    onCancel,
  } = props;

  const dialog = new Adw.MessageDialog({
    heading: title,
    body: description,
    transientFor: parent,
    modal: true,
  });

  dialog.add_response('cancel', cancelLabel);
  dialog.add_response('confirm', confirmLabel);

  dialog.set_response_appearance(
    'confirm',
    destructive ? Adw.ResponseAppearance.DESTRUCTIVE : Adw.ResponseAppearance.SUGGESTED,
  );

  dialog.set_default_response('confirm');
  dialog.set_close_response('cancel');

  dialog.connect('response', (_dialog: Adw.MessageDialog, response: string) => {
    if (response === 'confirm') {
      onConfirm?.();
    } else {
      onCancel?.();
    }
  });

  return dialog;
}

File: surface/widgets/appkit/components/widgets/feedback/AlertDialog.swift
// ============================================================
// Clef Surface AppKit Widget — AlertDialog
//
// Modal confirmation dialog with title, message, and action
// buttons. Wraps NSAlert for native macOS alert behavior.
// ============================================================

import AppKit

public class ClefAlertDialog {
    public var title: String = ""
    public var message: String = ""
    public var confirmLabel: String = "OK"
    public var cancelLabel: String = "Cancel"
    public var destructive: Bool = false
    public var onConfirm: (() -> Void)?
    public var onCancel: (() -> Void)?

    public func show(in window: NSWindow? = nil) {
        let alert = NSAlert()
        alert.messageText = title
        alert.informativeText = message
        alert.alertStyle = destructive ? .critical : .informational

        alert.addButton(withTitle: confirmLabel)
        alert.addButton(withTitle: cancelLabel)

        if destructive {
            alert.buttons.first?.hasDestructiveAction = true
        }

        let handler: (NSApplication.ModalResponse) -> Void = { [weak self] response in
            if response == .alertFirstButtonReturn {
                self?.onConfirm?()
            } else {
                self?.onCancel?()
            }
        }

        if let window = window {
            alert.beginSheetModal(for: window, completionHandler: handler)
        } else {
            let response = alert.runModal()
            handler(response)
        }
    }
}
